packages/bot-web-ui/src/components/dashboard/bot-builder/toolbox/toolbox.tsx

```diff
@@ -0,0 +1,165 @@
+import React from 'react';
+import classNames from 'classnames';
+import { Icon, Text } from '@deriv/components';
+import { localize } from '@deriv/translations';
+import { ToolboxItems } from './toolbox-items';
+import { connect } from '../../../../stores/connect';
+import RootStore from 'Stores/index.js';
+import SearchBox from './search-box';
+
+type TToolbox = {
+    hasSubCategory: (param: HTMLCollection) => boolean;
+    is_mobile: boolean;
+    is_search_loading: boolean;
+    is_toolbox_open: boolean;
+    onMount: (param?: React.RefObject<typeof ToolboxItems>) => void;
+    onSearch: () => void;
+    onSearchBlur: () => void;
+    onSearchClear: () => void;
+    onSearchKeyUp: () => void;
+    onToolboxItemClick: (category: ChildNode) => void;
+    onToolboxItemExpand: (index: number) => void;
+    onUnmount: () => void;
+    sub_category_index: number[];
+    toggleDrawer: () => void;
+    toolbox_dom: HTMLElement;
+};
+
+const Toolbox = ({
+    onMount,
+    onUnmount,
+    hasSubCategory,
+    is_mobile,
+    is_search_loading,
+    onSearch,
+    onSearchBlur,
+    onSearchClear,
+    onSearchKeyUp,
+    onToolboxItemClick,
+    onToolboxItemExpand,
+    sub_category_index,
+    toolbox_dom,
+}: TToolbox) => {
+    const toolbox_ref = React.useRef(ToolboxItems);
+    const [is_open, setOpen] = React.useState(true);
+
+    React.useEffect(() => {
+        onMount(toolbox_ref);
+        return () => onUnmount();
+    }, [onMount, onUnmount]);
```

Comment:
IINM we don't add functions in the dependency array unless they're memoized.

Reply:
In previous bot version we had these methods as useEffect dependency list, but with the new version we don't need it, so we will remove it. ✅